### ISSUE TITLE
Project: Aware is a sidecar to help you triage when you need a break

### ISSUE THREAD

**Issue by felipeschneider88** [microsoft/hack-together#76]
### Project name

mental healt aware

## The problem we're solving
These days is very common to see people struggling with depression and mental health. The _WHO (World Health Organization)_ said we [need to transform mental health](https://www.who.int/news/item/17-06-2022-who-highlights-urgent-need-to-transform-mental-health-and-mental-health-care)

## The solution?
Create a WebAPI to access the user's calendar and manage the coming events and reschedule if needed so he/she can take a small or long break.

## Features

- Ask the user how is feeling right now.
- Make a quick 20min meeting in the user calendar, so he/she can take a break.
- Block the calendar with an all-day appointment when the user says is not feeling well. 

### Repo URL

https://github.com/felipeschneider88/mental-health-aware

### Team members

felipeschneider88

**Comment by waldekmastykarz** [comment]
Awesome! Could you also submit the companion Office add-in? It would be awesome to see it in action end to end 😊

**Comment by waldekmastykarz** [comment]
Hey @felipeschneider88, would you mind recording a short video walkthrough of your hack? Any format will work. It's not required for the hackathon but it would help us showcase your work both internally at Microsoft and publicly in the community. 😊

**Comment by felipeschneider88** [comment]
Hello @waldekmastykarz , I will send the video tomorrow. I'm facing problems with the Office add-in to consume the web API so I will share the idea on the video. I'm not sure if I will fix it before the deadline and I prefer to share what I have

**Comment by waldekmastykarz** [comment]
Could you please share any more information about what's not working with the add-in? Perhaps we could help you fix the issue.

**Comment by felipeschneider88** [comment]
Yes, Today the app is built with 2 projects. One is the web API that interact with the Microsoft graph, it has the business logic on it.

The second is an Office add-in project that uses the SSO option to get the logged user. I was able to get information about the logged user in Excel. I was also able to get info from an URL that doesn't need auth 
``` javascript
const element = document.querySelector('#product-name');
fetch('https://testapi.jasonwatmore.com/products/1)
    .then(response => response.json())
    .then(data => element.innerHTML = data.name);
```

But when I tried to get a token from the SSO to call the web API It fails 

This is the code I'm using on the Excel Add-in:

``` javascript
async function getCalendar() {
  let userTokenEncoded = await OfficeRuntime.auth.getAccessToken({
    allowSignInPrompt: true,
    allowConsentPrompt: true,
    forMSGraphAccess: true,
  });  
  const element = document.getElementById("item-subject");
  let path = 'https://localhost:7068/calendar'
  $.ajax({
		url: path,
		method: "GET",
		headers: {
			"Authorization" : "Bearer " + userTokenEncoded
		},
		success: function(response) {
			element.innerHTML = response.json();
		}
	});
}
```

**Comment by waldekmastykarz** [comment]
Do I understand it correctly that your API is rejecting your token? Have you tried looking into the token at https://jwt.ms to see to which audience it's issued and if that audience is valid with your API?

**Comment by felipeschneider88** [comment]
I have uploaded the presentation video now, in a couple of hours I will be submitting the demo video too

**Comment by felipeschneider88** [comment]
**Introduction video**  
[![Introduction video](https://img.youtube.com/vi/TkHRsVpxN8I/0.jpg)](https://www.youtube.com/watch?v=TkHRsVpxN8IE)

**Demo video**  
[![Demo video](https://img.youtube.com/vi/aIbCKyoye2I/0.jpg)](https://www.youtube.com/watch?v=aIbCKyoye2I)
